Update properties.asciidoc (#2417)
vaadin/docs: articles/configuration/properties.asciidoc

|`false`

|`devmode.sessionSerialization.enabled`
|When session serialization is enabled, [classname]`UI` instances and registered [classname]`StreamResource` instances are serialized or deserialized when restarting the development server. Enabling the property allows, for example, access control information to be preserved during development, so that you don't need to log in again for each individual change. Only used in development mode.
|`false`

|`devmode.usageStatistics.enabled`
|Enables Vaadin to collect usage statistics that are used to guide development further. Statistics are collected based on features used in the application. No data is collected in production mode. Some usage statistics through the web browser. See the documentation for https://github.com/vaadin/vaadin-usage-statistics[the client-side collector] for information on how to opt out of that part of the usage statistics collection. Only used in development mode.
|`true`

|`disable.automatic.servlet.registration`
|Disable automatic registration of servlets that are needed for Vaadin application to work. If disabled, you should register Vaadin servlets yourself.
|`false`

|`disable-xsrf-protection`

> |Disbale cross-site request forgery protection. The protection is enabled by default, It is not recommended to disable protection, but you may want to disable it for a certain type of testing.

Reviewer: ```suggestion
|Disable cross-site request forgery protection. The protection is enabled by default. You should keep it enabled, but you may want to disable it for a certain type of testing.
```

|`false`

|`eagerServerLoad`

> |Enable the client-side bootstrap page should also include the initial UIDL JSON data fragment.

Reviewer: I wonder if readers know what UIDL is, and if we should avoid using that term. I’m not sure it’s explained anywhere.

Author: True, it is legacy. It was there, so didn't want to take it away either.

Author: I decided to remove it for now.

|`false`

|`frontend.hotdeploy`
|Enable development using the front-end development server instead of using an application bundle. Only applies to development mode.
|`false`

|`heartbeatInterval`
|UIs that are open on the client-side send a regular heartbeat to the server indicating that they are still alive, even when there is no ongoing user interaction. When the server doesn't receive a valid heartbeat from a given UI within a certain amount of time, it removes that UI from the session. The interval value is expressed in `seconds`. See also `closeIdleSessions`.
|300 seconds (5 minutes)

|`i18n.provider`

> |Fully qualified class name of I18N provider property. To use localization and translation of strings, the application should implement `I18NProvider` interface and define the class name in the property `i18n.provider`. See the <<{articles}/advanced/i18n-localization#, Localization>> documentation for more details.

Reviewer: ```suggestion
|Fully qualified name of internationalization provider class. To use localization and translation of strings, the application should implement `I18NProvider` interface and define the class name in the property `i18n.provider`. See the <<{articles}/advanced/i18n-localization#, Localization>> documentation for more details.
```

|`false`

|`productionMode`
|Sets the application to work in production mode. Disables most of the logged information that appears on the server and browser console to improve performance. Development mode JavaScript functions aren't exported. A `push` is given as a minified JavaScript file instead of a full-size one, and static resources are cached. See <<../production#,Deploying to Production>> for more information. Set to `true` when building application for public deployment.
|`false`

|`pushLongPollingSuspendTimeout`
|Timeout in `milliseconds ` for network request when using the long polling transport. If you have long polling enabled with a proxy with a timeout, you want `pushLongPollingSuspendTimeout` to be shorter than proxy timeout to make clients reconnect properly.
|`-1` (no timeout)

|`pushMode`
|Enable the server push. The permitted values are "disabled", "manual" or "automatic". See <<{articles}/advanced/server-push#, Server Push>> for more information.
|`disabled`

|`pushServletMapping`

> |Servlet mapping used for bidirectional ("push") client-server communication. Some Java application servers require special context e.g. for websockets and you can specify it here.

Reviewer: Apparently we should add `websockets?` to the vocabulary.
